### ISSUE TITLE
[Payment Request] [$250]  Mention - Avatar is not displaying on Concierge in mention list

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#41014]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.66-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4516979&group_by=cases:section_id&group_order=asc&group_id=306201&group_by=cases:section_id&group_order=asc&group_id=306201
**Email or phone of affected tester (no customers):** shussain+accoun1@applausemail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Open a chat
2. Press @ and check Concierge

## Expected Result:
Avatar should be displayed on Concierge in mention list

## Actual Result:
Avatar is not displaying on Concierge in mention list

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/dc9b332d-a255-4240-a01e-a035a0beeaf5

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~015c5e79bf2b62e7b7</li>
        <li>Upwork Job ID: 1783566739983851520</li>
        <li>Last Price Increase: 2024-04-25</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @allroundexperts</details>

**Comment by neil-marcellini** [comment]
Ok, taking a look. In general we can add external and the contributors can help us figure out if it's a backend problem.

**Comment by neil-marcellini** [comment]
Thanks @ZhenjaHorbach. Can you explain the root cause a bit more please? Why is it that we don't pass the accountID when creating the mention? It looks to me like we do.

**Comment by shahinyan11** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Mention - Avatar is not displaying on Concierge in mention list
### What is the root cause of that problem?
[Here](https://github.com/Expensify/App/blob/6060e6ffd5387975e65339344828141d83c9d642/src/libs/UserUtils.ts#L151C5-L151C101) we create default avatar if the `isDefaultAvatar(avatarSource)` is `true`
### What changes do you think we should make in order to solve the problem?
Remove [this](https://github.com/Expensify/App/blob/6060e6ffd5387975e65339344828141d83c9d642/src/libs/UserUtils.ts#L135C63-L135C104) check from condition
### What alternative solutions did you explore? (Optional)

**Comment by neil-marcellini** [comment]
Ok @ZhenjaHorbach I think I finally understand your proposal. I would say the root cause is this.
1. When we build the suggestions the source is set to this https://github.com/Expensify/App/blob/c6117097e247a8ca7c51b5d59d9d08a901b75162/src/pages/home/report/ReportActionCompose/SuggestionMention.tsx#L245
Previously it was this, and was [changed here](https://github.com/Expensify/App/commit/441d4953d0e1513bce57d2450ce83b2335fdb6fc#diff-ee52136d0310781e51a5cdc7f924e4e1ed6c7ca477e906f1953ae531e3f72f38L241-R242).
https://github.com/Expensify/App/blob/45cd559d2a4af67b8610a87bb472840da087cf80/src/pages/home/report/ReportActionCompose/SuggestionMention.tsx#L241

2. MentionSuggestions renders the Avatar here and doesn't pass an accountID
https://github.com/Expensify/App/blob/c6117097e247a8ca7c51b5d59d9d08a901b75162/src/components/MentionSuggestions.tsx#L82-L89

3. The avatar source is set here using UserUtils.getAvatar. The accountID is undefined
https://github.com/Expensify/App/blob/c6117097e247a8ca7c51b5d59d9d08a901b75162/src/components/Avatar.tsx#L89

4. That calls getDefaultAvatar https://github.com/Expensify/App/blob/c6117097e247a8ca7c51b5d59d9d08a901b75162/src/libs/UserUtils.ts#L151
5. Since there's not accountID, it incorrectly returns a default avatar
https://github.com/Expensify/App/blob/c6117097e247a8ca7c51b5d59d9d08a901b75162/src/libs/UserUtils.ts#L106


I'm wondering how we can test this. I can't reproduce on dev because the avatar is always blank for me. I'll try against the staging api instead of our dev api.

Or we could just revert [Use fallback user avatar in cases where the user is unknown to us](https://github.com/Expensify/App/pull/39229#top). @grgia what do you think since you were involved there?

**Comment by grgia** [comment]
@Kicu could you take a look at this?

**Comment by Kicu** [comment]
hey I'm the author of https://github.com/Expensify/App/pull/39229 and I took a look at this, few points from me in random order
 - this is a great description of root cause: https://github.com/Expensify/App/issues/41014#issuecomment-2078146075 - kudos to you @neil-marcellini 
 - I can easily reproduce on dev - on main you get what we call "FallbackAvatar" but if I pass accountID then I get concierge avatar
 - before this PR there was 20+ unneeded calls to `getAvatar()` that basically repeated logic, so my intention in PR 39229 was to keep pushing down `details.avatar` prop and only run `getAvatar()` once - IMO this simplifies the already complex avatar logic
 
Because of this I believe the best fix for this is to actually do pass `accountID` prop correctly. This PR seems to do just that (https://github.com/Expensify/App/pull/41034).

I tested locally on my dev the same fix, and it looks good
### before (main)
<img width="500" src="https://github.com/Expensify/App/assets/3929868/7754199a-7b0a-4df3-bb30-2e758adb9215" />

### after passing accountID
<img width="500" src="https://github.com/Expensify/App/assets/3929868/045cafad-1bea-47ad-9a67-6a8ac37f41e0" />

In general we only need `accountID` to be able to display the "pretty" svg versions of: default avatars, ConciergeAvatar and NotificationsAvatar. Otherwise `.avatar` from user details is always correct.
In addition I can supply a separate PR that adds the new `accountID` prop in a few places where Avatar is used, because it seems I might miss them.


@grgia

**Comment by grgia** [comment]
@Kicu are you able to open a PR fix for this as a followup in that case?

**Comment by Kicu** [comment]
@grgia we can use this fix by @neil-marcellini I believe - it's already created:
https://github.com/Expensify/App/pull/41034

**Comment by allroundexperts** [comment]
@neil-marcellini Can you please assign this to me since I reviewed the PR?

**Comment by mountiny** [comment]
Ended up reverting the PR to unblock the deploy, assign @Kicu to address this in the second iteration

**Comment by allroundexperts** [comment]
The PR was closed and the offending PR was reverted. As such, we don't need to wait for the 7 days for the payment.

**Comment by neil-marcellini** [comment]
Sounds like this is all done then. Please correct me if I'm wrong.

**Comment by allroundexperts** [comment]
Can I have a payment summary @neil-marcellini? Thanks!

**Comment by JmillsExpensify** [comment]
Re-opening for payment summary.

**Comment by neil-marcellini** [comment]
Assigning a bug zero member to handle payment

**Comment by strepanier03** [comment]
### Payment Summery

- $250 - @allroundexperts via Manual Requests in New Expensify

@JmillsExpensify - Payment request incoming.

**Comment by JmillsExpensify** [comment]
$250 approved for @allroundexperts
